src/OpenFeature.Contrib.ConfigCat/ConfigCatProvider.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Threading.Tasks;
+using ConfigCat.Client;
+using ConfigCat.Client.Configuration;
+using OpenFeature.Constant;
+using OpenFeature.Model;
+
+namespace OpenFeature.Contrib.ConfigCat
+{
+    /// <summary>
+    /// ConfigCatProvider is the .NET provider implementation for the feature flag solution ConfigCat.
+    /// </summary>
+    public class ConfigCatProvider : FeatureProvider
+    {
+        private const string Name = "ConfigCat Provider";
+        internal readonly IConfigCatClient Client;
+
+        /// <summary>
+        /// Creates new instance of <see cref="ConfigCatProvider"/>
+        /// </summary>
+        /// <param name="sdkKey">SDK Key to access the ConfigCat config.</param>
+        /// <param name="configBuilder">The action used to configure the client.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="sdkKey"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="sdkKey"/> is an empty string or in an invalid format.</exception>
+        public ConfigCatProvider(string sdkKey, Action<ConfigCatClientOptions> configBuilder = null)
+        {
+            Client = ConfigCatClient.Get(sdkKey, configBuilder);

```

Comment:
One more remark. `ConfigCatClient` implements `IDisposable` and instances of such types should usually be disposed by the consumer as soon as they don't need them any longer.

I'm not sure though how big an issue is this in the case of OpenFeature, what are your guidelines regarding resource management. Probably it's not an actual issue as when we were implementing `ConfigCatClient`, we paid careful attention that client instances can be collected by the GC when no strong references to them are left.

Reply:
For what it is worth, I have implemented the `IDisposable` interface in the provider class to handle this correctly.